platform/security/certificate/security-certificate-keystoreeditor/src/main/java/org/codice/ddf/security/certificate/keystore/editor/KeystoreEditor.java

```diff
@@ -266,6 +280,96 @@ public void addTrustedCertificate(String alias, String keyPassword, String store
+                trustStoreFile = Paths.get(ddfHomePath.toString(), trustStoreFile.toString());
+            }
+            String keyStorePassword = SecurityConstants.getTruststorePassword();
+            OutputStream fos = Files.newOutputStream(trustStoreFile);
+            trustStore.store(fos, keyStorePassword.toCharArray());
+        } catch (IOException | GeneralSecurityException e) {
+            LOGGER.error("Unable to add certificate(s) to trust store from URL: {}",
+                    (decodedUrl != null) ? decodedUrl : url, e);
+        } finally {
+            IOUtils.closeQuietly(socket);
+        }
+        return resultList;
+    }
+
+    @Override
+    public List<Map<String, Object>> showCertificateFromUrl(String url) {
+        List<Map<String, Object>> certificates = new ArrayList<>();
+        SSLSocket socket = null;
+        String decodedUrl = null;
+        try {
+            decodedUrl = new String(Base64.getDecoder()
+                    .decode(url), "UTF-8");
+            socket = getNonVerifyingSslSocket(decodedUrl);
+            socket.startHandshake();
+            X509Certificate[] peerCertificateChain = (X509Certificate[]) socket.getSession()
+                    .getPeerCertificates();
+            for (X509Certificate certificate : peerCertificateChain) {
+                Map<String, Object> certMap = new HashMap<>();
+                certMap.put("type", certificate.getType());
+                certMap.put("issuerDn", certificate.getIssuerDN());
+                certMap.put("subjectDn", certificate.getSubjectDN());
+                certMap.put("extendedKeyUsage", certificate.getExtendedKeyUsage());
+                certMap.put("notAfter", certificate.getNotAfter()
+                        .toString());
+                certMap.put("notBefore", certificate.getNotBefore()
+                        .toString());
+                certMap.put("thumbprint", DigestUtils.sha1Hex(certificate.getEncoded()));
+                certificates.add(certMap);
+            }
+        } catch (IOException | GeneralSecurityException e) {
+            LOGGER.error("Unable to parse certificate from URL: {}",
+                    (decodedUrl != null) ? decodedUrl : url, e);
+        } finally {
+            IOUtils.closeQuietly(socket);
+        }
+
+        return certificates;
+    }
+
+    SSLSocket getNonVerifyingSslSocket(String decodedUrl)
+            throws NoSuchAlgorithmException, KeyManagementException, KeyStoreException,
+            UnrecoverableKeyException, IOException {
+        URL httpsUrl = new URL(decodedUrl);
+        SSLContext sslContext = SSLContext.getInstance("TLS");
+        TrustManager tm = new NonVerifyingTrustManager();
+        sslContext.init(null, new TrustManager[] {tm}, null);
+        SSLConnectionSocketFactory sslSocketFactory = new NonVerifyingSslSocketFactory(sslContext);
+        return (SSLSocket) sslSocketFactory.connectSocket(30000, null,
```

Comment:
I'm not sure at all how long it might take. 30 seconds was an arbitrary choice by me.